src/Klarna/Rest/Resource.php

```diff
@@ -139,9 +139,38 @@ public function fetch()
      */
     protected function request($method, $url, array $headers = [], $body = '')
     {
-        $request = $this->connector->createRequest($url, $method, $headers, $body);
+        $debug = getenv('DEBUG_SDK') || defined('DEBUG_SDK');
 
-        return new ResponseValidator($this->connector->send($request));
+        $request = $this->connector->createRequest($url, $method, $headers, $body);
+        if ($debug) {
+            $clientConfig = $this->connector->getClient()->getConfig();
+            $baseUri = isset($clientConfig['base_uri']) ? $clientConfig['base_uri'] : '';
+            $debugHeaders = $request->getHeaders();
+            $debugHeaders = json_encode($debugHeaders);
```

Comment:
Might want to obfuscate the Authorization header if it is included here.

Reply:
Authorization header is not here. This header will be added only later by `connector`